src/library/scala/collection/immutable/Vector.scala

```diff
@@ -110,17 +110,15 @@ final class Vector[+A] private[immutable] (private[collection] val startIndex: I
 
   @throws[IndexOutOfBoundsException]
   def apply(index: Int): A = {
-    val idx = checkRangeConvert(index)
+    val idx = rangeConvert(index)
```

Comment:
I assume the range check was factored out to a separate method for performance. Could we keep that structure?

Reply:
My initial intent of refactoring was to have different messages for methods, like this
* apply: `"No element at $index since bound is..."`  (However, it was accidentally same for `updateAt` 😨 )
* updateAt: `"Can't update at $index since bound is ..."`

However, since all messages should be `"XX is out of bounds (min 0, max YY)"` as lrytz designed, we can keep the stcture.

- [ ] Keep `checkRangeConvert` and add message.

However, since